### PR TITLE
[NTOS:MM] Correctly handle SizeOfRawData

### PR DESCRIPTION
**Pull request by ThFabba**
JIRA issue: [CORE-17284](https://jira.reactos.org/browse/CORE-17284)

### CODE REVIEW

File: ntoskrnl/mm/section.c [hunk 1]
```diff
@@ -754,7 +758,7 @@ NTSTATUS NTAPI PeFmtCreateSection(IN CONST VOID * FileHeader,
         pssSegments[i].Protection = SectionCharacteristicsToProtect[nCharacteristics >> 28];
         pssSegments[i].WriteCopy = !(nCharacteristics & IMAGE_SCN_MEM_SHARED);
 
-        if(pishSectionHeaders[i].Misc.VirtualSize == 0 || pishSectionHeaders[i].Misc.VirtualSize < pishSectionHeaders[i].SizeOfRawData)
+        if(pishSectionHeaders[i].Misc.VirtualSize == 0)
```
Comment (HBelusca) [inline ntoskrnl/mm/section.c:761]: Check how this is done in freeldr and in Alex's bootlib:
https://git.reactos.org/?p=reactos.git;a=blob;f=boot/freeldr/freeldr/lib/peloader.c;hb=895dccd4ac7cc9c8dcb4a679e34fe0b50c560b2e#l819
https://git.reactos.org/?p=reactos.git;a=blob;f=boot/environ/lib/misc/image.c;hb=895dccd4ac7cc9c8dcb4a679e34fe0b50c560b2e#l967
(Basically, if SizeOfRawData > VirtualSize then set SizeOfRawData = VirtualSize). Indeed, the raw data (and its size) size might have been rounded up to the next section alignment, but we know that its meaningful contents, when loaded in memory, must not be larger than VirtualSize.

Comment (ThFabba) [reply ntoskrnl/mm/section.c:761]: I'm not sure what change you're suggesting, if any.
My tests cover `SizeOfRawData > VirtualSize`, `SizeOfRawData < VirtualSize`, `SizeOfRawData = 0`, and they all pass with the code as presented in this PR.

Comment (HBelusca) [reply ntoskrnl/mm/section.c:761]: Just showing how the sections loop contents should look like...

File: modules/rostests/apitests/ntdll/NtMapViewOfSection.c [hunk 2]
```diff
@@ -1473,6 +1473,370 @@ Test_SectionContents(BOOL Relocate)
...
+        ok_ntstatus(Status, STATUS_INVALID_IMAGE_FORMAT);
+    }
+    else
+    {
+        ok_ntstatus(Status, STATUS_SUCCESS);
+    }
+
+    if (NT_SUCCESS(Status))
+    {
+        /* Map the section with  */
```
Comment (learn-more) [inline modules/rostests/apitests/ntdll/NtMapViewOfSection.c:1771]:
```suggestion
        /* Map the section with PAGE_READWRITE */
```

**Review by learn-more — APPROVED** [review]

**Review by tkreuzer — APPROVED** [review]

**Pull request merged**